### PR TITLE
Add an additional diagonal hybrid domain test

### PR DESCRIPTION
**Pull request by bischtob**
**context**
Hybrid domain integration tests currently test horizonal advection and vertical advection separately. This PR adds an additional diagonal advection test against an analytical solution.

### CODE REVIEW

**Comment by bischtob** [comment]
bors try

**Comment by bischtob** [comment]
@Zhengyu-Huang, this is a 2D advection test case with analytical solution. Please check it out for your purposes.

File: test/hybrid.jl [hunk 1]
```diff
@@ -107,3 +112,60 @@ end
...
+
+    using OrdinaryDiffEq
+    U = Fields.FieldVector(h = h_init(0.5, 0.5))
+    Δt = 0.01
+    t_end = 1.0
+    prob = ODEProblem(rhs!, U, (0.0, t_end))
+    sol = solve(prob, SSPRK33(), dt = Δt)
+
+    h_end = h_init(0.5 - t_end, 0.5 - t_end)
+    @test norm(h_end .- sol.u[end].h) ≤ 5e-2
```
Comment (jakebolewski) [inline test/hybrid.jl:170]: should we check the error of the vertical / horizontal only components individually as well as the overall 2D solution?

**Review by jakebolewski — APPROVED** [review]

**Review by Zhengyu-Huang — APPROVED** [review]
I can run it! All tests pass, great job!

**Pull request merged**